bin/analyze

```diff
@@ -0,0 +1,367 @@
+        COMMIT_CATEGORY, ISSUE_CATEGORY and PULL_REQUEST_CATEGORY.
+
+    :returns data: A dictionary of fetched items, segregated by category.
+    """
+
+    if api_token is None:
+        api_token = []
+
+    logging.info("Fetching data")
+    data = {
+        COMMIT_CATEGORY: [],
+        ISSUE_CATEGORY: [],
+        PULL_REQUEST_CATEGORY: []
+    }
+
+    for category in categories:
+        if category == COMMIT_CATEGORY:
+
+            repo_uri = GITHUB_URI + owner + '/' + repository + '.git'
+            print(repo_uri)
+            git = Git(uri=repo_uri, gitpath='tmp')
+            items = list(git.fetch(category=category))
+
+        else:
+            github = GitHub(owner=owner, repository=repository,
+                            api_token=api_token)
+            items = list(github.fetch(category=category))
+
+        data[category] = items
+
+    data['issue'] = [item for item in data['issue'] if 'pull_request' not in item]
+    return data
+
+
+def main():
+    """
+    A script for evaluating the values of Evolution WG metrics on user data.
+    Data is fetched using Perceval's back-ends and metrics are evaluated on it.
+
+    The results of the analysis can be output in several different formats.
+    Currently, json, pdf and images are supported.
+
+    The script can be run via the command line:
+        $ analyze -r chaoss/wg-evolution
+
+    Examples:
+    --------
+
+    * Run all commit metrics on chaoss/wg-evolution, starting from 2018-01-01
+    and using the api-token xxxx:
+        $ analyze -r chaoss/wg-evolution -s 2018-01-01 -t xxxx  -cat commit
+
+    * Run all metrics on chaoss/grimoirelab-perceval and generate a pdf report
+    of the results:
+        $ analyze -r chaoss/grimoirelab-perceval -o pdf
+
+    * Run commit metrics on chaoss/wg-evolution considering only non-empty commits
+    created on the master branch
+        $ analyze -r chaoss/wg-evolution -cat commit -c EmptyExclude MasterInclude
+    """
+
+    args = parse_args()
+
+    # owner and repo parsing
+    owner, repo = args.repo.split('/')
+
+    # date parsing
+    since = None
+    until = None
+    if args.since:
+        since = datetime.strptime(args.since, "%Y-%m-%d")
+    if args.until:
+        until = datetime.strptime(args.until, "%Y-%m-%d")
+
+    date_range = (since, until)
+
+    # conds parsing
+    conds_options = {
+        'MergeExclude': MergeExclude(),
+        'EmptyExclude': EmptyExclude(),
+        'MasterInclude': MasterInclude()
+    }
+
+    conds = [conds_options[option] for option in args.conds]
+
+    # is_code parsing
+    is_code_options = {
+        'Naive': Naive(),
+        'PostfixExclude': PostfixExclude(args.postfixes_to_exclude),
+        'DirExclude': DirExclude(args.dirs_to_exclude)
+    }
+
+    is_code = [is_code_options[option] for option in args.is_code]
+
+    configure_logging(args.debug)
+
+    # fetching data
+    items = fetch_data(owner, repo, [args.api_token], args.categories)
```

Comment:
An error was thrown here, changing this line to `items = fetch_data(owner, repo, args.api_token, args.categories)` solved the problem

Reply:
I am having [this issue](https://github.com/chaoss/grimoirelab-perceval/issues/546). When I use the brackets, I get the right output.
If I remove the brackets, I get this:
```bash
(wg_imp) Polaris000@thinkpad:~/Documents/GSOC/wg-gmd$ analyze -r Polaris000/practise -o json -cat issue pull_request -t xxxx
[2019-08-17 17:43:46,277] - Fetching data
[2019-08-17 17:43:46,678] - Rate limit not initialized: 401 Client Error: Unauthorized for url: https://api.github.com/rate_limit
[2019-08-17 17:43:46,939] - Rate limit not initialized: 401 Client Error: Unauthorized for url: https://api.github.com/rate_limit
[2019-08-17 17:43:47,179] - Rate limit not initialized: 401 Client Error: Unauthorized for url: https://api.github.com/rate_limit
...
..
requests.exceptions.HTTPError: 401 Client Error: Unauthorized for url: https://api.github.com/rate_limit
```

Reply:
yes, you are right, I was using an old version of perceval.
Could you update the requirements.txt to include:
- grimoirelab-perceval (probably >=0.12.18)
- grimoirelab-toolkit (probably >=0.1.9)
- fpdf (probably >=1.7.2)

Reply:
sure! Thanks.